engine/player/covenant.cpp

```diff
@@ -449,28 +451,44 @@ void covenant_state_t::register_options( player_t* player )
           this, std::placeholders::_1, std::placeholders::_2, std::placeholders::_3 ) ) );
 }
 
-report::sc_html_stream& covenant_state_t::generate_report( report::sc_html_stream& root ) const
+unsigned covenant_state_t::get_covenant_ability_spell_id( bool generic ) const
```

Comment:
You could just return the found spell data here instead of the id, since you seem to be using spell data later on anyway.